lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -227,6 +258,88 @@ public APIGatewayProxyResponseEvent handleRequest(
         }
     }
 
+    private static Optional<InheritedIdentityJwtClaim> getInheritedIdentityClaim(
+            JWTClaimsSet claimsSet) throws ParseException {
+        return Optional.ofNullable(
+                        OBJECT_MAPPER.convertValue(
+                                claimsSet.getJSONObjectClaim(CLAIMS_CLAIM), JarClaims.class))
+                .map(JarClaims::userInfo)
+                .map(JarUserInfo::inheritedIdentityClaim);
+    }
+
+    @Tracing
+    private void validateAndStoreHMRCInheritedIdentity(
+            String userId,
+            InheritedIdentityJwtClaim inheritedIdentityJwtClaim,
+            JWTClaimsSet claimsSet,
+            IpvSessionItem ipvSessionItem)
+            throws RecoverableJarValidationException, ParseException {
+        try {
+            // Validate
+            var inheritedIdentityJwtList =
+                    validateInheritedIdentityJwtList(inheritedIdentityJwtClaim);
+
+            var signedInheritedIdentityJWT = SignedJWT.parse(inheritedIdentityJwtList.get(0));
+            var inheritedIdentityCriConfig = configService.getCriConfig(HMRC_MIGRATION_CRI);
+
+            verifiableCredentialJwtValidator.validate(
+                    signedInheritedIdentityJWT, inheritedIdentityCriConfig, userId);
+            LogHelper.logMessage(Level.INFO, "Migration VC successfully validated");
+
+            // Store
+            try {
+                verifiableCredentialService.persistUserCredentials(
+                        signedInheritedIdentityJWT, HMRC_MIGRATION_CRI, userId);
+                ipvSessionItem.addVcReceivedThisSession(signedInheritedIdentityJWT.serialize());
+                ipvSessionService.updateIpvSession(ipvSessionItem);
+                LogHelper.logMessage(Level.INFO, "Migration VC successfully persisted");
+            } catch (VerifiableCredentialException e) {
+                throw buildRecoverableException(
+                        OAuth2Error.SERVER_ERROR
+                                .setDescription("Failed to persist inherited identity JWT - ")
+                                .appendDescription(e.getMessage()),
+                        claimsSet);
+            }
+        } catch (VerifiableCredentialException | ParseException e) {
+            throw buildRecoverableException(
+                    OAuth2Error.INVALID_REQUEST_OBJECT
+                            .setDescription("Inherited identity JWT failed to validate - ")
+                            .appendDescription(e.getMessage()),
+                    claimsSet);
+        } catch (JarValidationException e) {
+            throw buildRecoverableException(e.getErrorObject(), claimsSet);
+        }
```

Comment:
If you separate this into two methods, I think you can avoid the nested try/catch? i.e. `validateHmrcInheritedIdentity` and `storeHmrcInheritedIdentity` (the 'and' in this method name hints at it too!)